### PR TITLE
feat: Add support for RBS inline type annotations

### PR DESCRIPTION
**Pull request by pcasaretto**
## Summary

This PR adds support for recognizing RBS (Ruby Signature) inline type annotation comments in Ruby code. This enables better editor support for Ruby's type annotations by distinguishing them from regular comments in the AST.

## Motivation

RBS is Ruby's official type signature language, and increasingly Ruby developers are using inline RBS comments to annotate their code with type information:

```ruby
#: (Integer) -> String
def to_hex(num)
  num.to_s(16)
end

x = 42 #: Integer
```

Currently, tree-sitter-ruby treats these as regular comments, making it impossible for editors to provide enhanced support for type annotations (different highlighting, hover information, etc.).

## Implementation

This implementation takes a **minimal approach** - it recognizes RBS comment patterns without parsing the type syntax itself:

- `#:` comments are recognized as `rbs_type_comment` tokens
- `#|` comments (multi-line continuations) are recognized as `rbs_continuation_comment` tokens
- Regular `#` comments remain unchanged

### Why not parse the full RBS syntax?

1. **Separation of concerns**: RBS is effectively a different language. Tools that need to understand RBS types can extract the comment content and parse it with a dedicated RBS parser.
2. **Simplicity**: Full RBS syntax would require 200+ additional grammar rules
3. **Maintenance**: Keeping them separate means each parser can evolve independently

## Changes

- **grammar.js**: Added `rbs_type_comment` and `rbs_continuation_comment` token rules
- **queries/highlights.scm**: Added highlighting as `@type.annotation` (vs `@comment` for regular comments)
- **test/corpus/rbs_annotations.txt**: Added comprehensive test coverage with 11 test cases

## Testing

All tests pass (301 total, including 11 new RBS tests):
- ✅ Method signatures
- ✅ Variable annotations
- ✅ Type assertions
- ✅ Multi-line signatures
- ✅ Complex types (unions, generics, optionals)
- ✅ Mixed regular and RBS comments

## Impact

- **Non-breaking**: Existing Ruby parsing is completely unchanged
- **Editor-friendly**: Provides foundation for enhanced type annotation support
- **Future-proof**: Can be extended if needed without breaking changes

## Related Work

- [tree-sitter-rbs](https://github.com/joker1007/tree-sitter-rbs) - Standalone parser for .rbs files
- [RBS documentation](https://github.com/ruby/rbs/blob/master/docs/syntax.md)
- [Sorbet RBS support](https://sorbet.org/docs/rbs-support)

### CODE REVIEW

File: test/corpus/rbs_annotations.txt [hunk 1]
```diff
@@ -0,0 +1,231 @@
+=======================
```
Comment (Morriar) [inline test/corpus/rbs_annotations.txt:1]: Could you add a test showing the behavior if the `#:` is found inside a string:

```rb
puts "#: Hello, World"
```

Comment (pcasaretto) [reply test/corpus/rbs_annotations.txt:1]: Done!

Comment (pcasaretto) [reply test/corpus/rbs_annotations.txt:1]: Added some more edge cases involving strings.

**Review by Morriar — APPROVED** [review]

**Comment by Morriar** [comment]
This is a cool change, how do we get someone from the team to review/merge it? @amaanq I see you've been active in this repo recently 🙏
